### PR TITLE
[kubeapps-apis] Add direct-helm GetAvailablePackageDetail 

### PR DESCRIPTION
**Pull request by antgamdia**
### Description of the change

This PR just implements the `GetAvailablePackageDetail` method in the direct-helm plugin.

### Benefits

We will have a brand new `GetAvailablePackageDetail` operation up and running. 

### Possible drawbacks

N/A

### Applicable issues
  - related #2993
  - depends on https://github.com/kubeapps/kubeapps/pull/3022
  - depends on https://github.com/kubeapps/kubeapps/pull/3033

### Additional information

(Draft until we merge the other PRs so that it becomes easier to review; didn't want to push to the repo)

### CODE REVIEW

**Comment by absoludity** [comment]
> (Draft until we merge the other PRs so that it becomes easier to review; didn't want to push to the repo)

You can target your other branch in situations like this, so reviews can go ahead with just the relevant diff. It's a bit of a pain, but works OK. When the preceding branch lands, GH will automatically update the target of this PR back to the main branch, but you still often need to rebase, removing the extra commits.

File: cmd/kubeapps-apis/plugins/helm/packages/v1alpha1/server.go
Comment (absoludity) [inline cmd/kubeapps-apis/plugins/helm/packages/v1alpha1/server.go:289]: Is the description required? It's listed as optional at https://helm.sh/docs/topics/charts/#the-chartyaml-file

File: cmd/kubeapps-apis/plugins/helm/packages/v1alpha1/server.go
Comment (absoludity) [inline cmd/kubeapps-apis/plugins/helm/packages/v1alpha1/server.go:291]: I think we should just leave LongDescription blank for helm charts, which only have a single description field in the Chart.yaml? If we copy it like this, a client would display both, I'd think?

File: cmd/kubeapps-apis/plugins/helm/packages/v1alpha1/server.go
Comment (absoludity) [inline cmd/kubeapps-apis/plugins/helm/packages/v1alpha1/server.go:282]: Also listed as optional at https://helm.sh/docs/topics/charts/#the-chartyaml-file

File: cmd/kubeapps-apis/plugins/helm/packages/v1alpha1/server.go
Comment (absoludity) [inline cmd/kubeapps-apis/plugins/helm/packages/v1alpha1/server.go:293]: Same, not required at https://helm.sh/docs/topics/charts/#the-chartyaml-file

File: cmd/kubeapps-apis/plugins/helm/packages/v1alpha1/server.go
Comment (absoludity) [inline cmd/kubeapps-apis/plugins/helm/packages/v1alpha1/server.go:324]: AppVersion is also optional at https://helm.sh/docs/topics/charts/#the-chartyaml-file so I assume it can be optional in our cache.

File: cmd/kubeapps-apis/plugins/helm/packages/v1alpha1/server.go
Comment (absoludity) [inline cmd/kubeapps-apis/plugins/helm/packages/v1alpha1/server.go:329]: Readme file is optional at https://helm.sh/docs/topics/charts/#the-chart-file-structure

File: cmd/kubeapps-apis/plugins/helm/packages/v1alpha1/server.go
Comment (absoludity) [inline cmd/kubeapps-apis/plugins/helm/packages/v1alpha1/server.go:339]: values.schema.json also optional at https://helm.sh/docs/topics/charts/#the-chart-file-structure

File: cmd/kubeapps-apis/plugins/helm/packages/v1alpha1/server_test.go
Comment (absoludity) [inline cmd/kubeapps-apis/plugins/helm/packages/v1alpha1/server_test.go:430]: I guess you could add failing tests here to show that it should succeed without certain fields (ie. readme, icon, maintainers) then fix it for them to pass :)

**Review by absoludity — APPROVED** [review]
Excellent, thanks Antonio. A couple of points about required vs optional fields to see what you think, but otherwise +1

File: cmd/kubeapps-apis/plugins/helm/packages/v1alpha1/server.go
Comment (antgamdia) [reply cmd/kubeapps-apis/plugins/helm/packages/v1alpha1/server.go:289]: I forgot to verify it! Last week I thought: ok, let's put every param as mandatory and later on I'll look them up in the helm code or website... but...ouch, I never did.
Thanks for the pointer!

File: cmd/kubeapps-apis/plugins/helm/packages/v1alpha1/server.go
Comment (antgamdia) [reply cmd/kubeapps-apis/plugins/helm/packages/v1alpha1/server.go:291]: Ah, that makes sense. I just set both so that we can decide later. But, we better avoid duplicated fields :P

File: cmd/kubeapps-apis/plugins/helm/packages/v1alpha1/server_test.go
Comment (antgamdia) [reply cmd/kubeapps-apis/plugins/helm/packages/v1alpha1/server_test.go:430]: TDD is great for these fixes :P Added a `TestIsValidChart` test suite

**Comment by absoludity** [comment]
Great, good to go when the conflict is resolved.

**Pull request merged**
